cloud-sql-proxy/cloud-sql-proxy.sh

```diff
@@ -476,11 +564,13 @@ function update_master() {
 
   install_cloud_sql_proxy
   start_cloud_sql_proxy
-  configure_sql_client
 
   if [[ $ENABLE_CLOUD_SQL_METASTORE == "true" ]]; then
-    install_mysql_cli
-    initialize_metastore_db
+    install_db_cli
+
+    # Retry as there may be failures due to race condition
+    run_with_retries initialize_metastore_db
```

Comment:
What's the race condition? Should we call it only on m-0?

Reply:
I don't see an answer or change for this comment.

Reply:
forgot to click the comment button earlier. 

The race condition is when all master nodes trying to create database user and database. From testing it seems postgres is more prone to this issue. There was a PR https://github.com/GoogleCloudDataproc/initialization-actions/pull/288 a while back trying to solve this issue, not sure why it is reverted, maybe inadvertently.  retry works here as the retry only checks whether db user or database exists so it is a fast check. only the node that creates the database gets to run hive --initSchema.
Alternatively we could try to pick a specific master node, but in case some master node is faster than others and only a few master nodes, retry works better IMO